e2e-tests/gef/cypress/integration/unissued-facility-change-from-preview-AIN.spec.js

```diff
@@ -0,0 +1,258 @@
+
+      aboutFacilityUnissued.issueDateDay().should('have.value', '');
+      aboutFacilityUnissued.issueDateMonth().should('have.value', '');
+      aboutFacilityUnissued.issueDateMonth().should('have.value', '');
+      aboutFacilityUnissued.coverStartDateDay().should('have.value', '');
+      aboutFacilityUnissued.coverStartDateMonth().should('have.value', '');
+      aboutFacilityUnissued.coverStartDateYear().should('have.value', '');
+      aboutFacilityUnissued.coverEndDateDay().should('have.value', '');
+      aboutFacilityUnissued.coverEndDateMonth().should('have.value', '');
+      aboutFacilityUnissued.coverEndDateYear().should('have.value', '');
+    });
+
+    it('update facilities later link should take you back to application preview', () => {
+      applicationPreview.unissuedFacilitiesReviewLink().click();
+      unissuedFacilityTable.updateIndividualFacilityButton(0).click();
+
+      aboutFacilityUnissued.issueDateDay().type(oneMonthDay);
+      aboutFacilityUnissued.issueDateMonth().type(oneMonthMonth);
+      aboutFacilityUnissued.issueDateYear().type(oneMonthYear);
+
+      aboutFacilityUnissued.shouldCoverStartOnSubmissionNo().click();
+      aboutFacilityUnissued.coverStartDateDay().type(twoMonthsDay);
+      aboutFacilityUnissued.coverStartDateMonth().type(twoMonthsMonth);
+      aboutFacilityUnissued.coverStartDateYear().type(twoMonthsYear);
+
+      aboutFacilityUnissued.coverEndDateDay().type(threeMonthsOneDayDay);
+      aboutFacilityUnissued.coverEndDateMonth().type(threeMonthsOneDayMonth);
+      aboutFacilityUnissued.coverEndDateYear().type(threeMonthsOneDayYear);
+      aboutFacilityUnissued.continueButton().click();
+
+      unissuedFacilityTable.successBanner().contains(`${unissuedFacilitiesArray[0].name} is updated`);
+      unissuedFacilityTable.rows().should('have.length', unissuedFacilitiesArray.length - 1);
+      unissuedFacilityTable.continueButton().should('not.exist');
+      unissuedFacilityTable.updateFacilitiesLater().click();
+    });
+
+    it('preview review facility stage has correct headers and shows 1 updated facilities', () => {
+      applicationPreview.reviewFacilityStage().contains('Review facility stage');
+
+      applicationPreview.updatedUnissuedFacilitiesHeader().contains('The following facility stages have been updated to issued:');
+      applicationPreview.updatedUnissuedFacilitiesList().contains(unissuedFacilitiesArray[0].name);
+      applicationPreview.unissuedFacilitiesReviewLink().should('not.exist');
+      applicationPreview.submitButtonPostApproval().should('exist');
+    });
+
+    it('facility table should have change links on the changed to issued facilities', () => {
+      // not be able to change 1 name, but can change to issued
+      applicationPreview.facilitySummaryListRowValue(0, 0).contains(MOCK_FACILITY_FOUR.name);
+      applicationPreview.facilitySummaryListRowAction(0, 0).should('have.value', '');
```

Comment:
This is probably me not understanding the UX/design, but the comment here - does this mean that an "Edit name" button _could_ appear, but here we are testing that the "row action" element is not rendered?

If so, should we be using `.should('not.exist');` instead of asserting there is no value?

Reply:
technically speaking, it is set to '' because of the mapping function.  Not sure if not.exist will work, can test it out or leave it as ''?

summaryItems = [
      ...(unissuedHref
        ? [
          {
            href: unissuedHref,
            /*  */
            text: `${unissuedShow ? 'Change' : ''}`,
            visuallyHiddenText: item.label,
          },
        ]
        : []),
    ];

Reply:
.should('not.exist'); does not work because of ^